### ISSUE TITLE
Sidebar aria-label is sometimes missing

### ISSUE THREAD

**Issue by ravicodelabs** [quarto-dev/quarto-cli#7059]
### Bug description

For screen readers, the Quarto sidebar adds a navigation landmark, however it does not always include a unique label. For instance:

- Our [webinars listing page](https://datascijedi.org/jedi-corner) navigation seems to do this automatically, e.g. if you inspect the pagination button you will see `<nav id="listing-pagination" class="listing-pagination" aria-label="Page Navigation">`
- Our [activities](https://datascijedi.org/activities) page navigation is different and does automatically add the aria-label, e.g. if you inspect the link to the next subpage you will see only `<nav class="page-navigation">`. 

### Steps to reproduce

See live example in bug description. 

### Expected behavior

An `aria-label` should be added whenever there are `nav` components used on the website. 

### Actual behavior

See bug description. 

### Your environment

Mac/Linux

### Quarto check output

Quarto v1.4.388 (but also occurred on earlier versions)
